libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/internal/loader/ResourceLoader.kt

```diff
@@ -0,0 +1,7 @@
+package com.mapbox.navigation.ui.shield.internal.loader
+
+import com.mapbox.bindgen.Expected
+
+internal fun interface ResourceLoader<Argument, Resource> {
```

Comment:
There is already a `ResourceLoader` in `com.mapbox.navigation.ui.utils.internal.resource`. Can we reuse that one? I find 2 classes with the same quite an abstract name confusing.

Reply:
No. They are not the same. The utils `ResourceLoader` is a global facade over the native TileStore, and it acts more like Downloader than a local loader (maybe `ResourceDownloader` would be a better name for it). 
This `ResourceLoader` is an **internal** interface. It offers an abstraction for a generic, async loader and helps break tight coupling between the `RoadShieldContentManager` and concrete loaders.
I am aware that the name clashes with the utils version. We can shorten it to just a `Loader` if it helps. Furthermore, I would like to introduce this abstraction here first and later, if we find it useful elsewhere, move it to the utils module as well.

Reply:
They should have different names. Especially considering the fact they are interfaces. So you can't even just look at the class code and understand the difference.
Maybe the utils one should be `TileStoreResourcesLoader` or sth like that. Since it uses Common classes in the API anyway.

Reply:
Since they are internal interfaces they don't have to have different names. 
Anyway, I've renamed it to just `Loader`.

Reply:
> Since they are internal interfaces they don't have to have different names.

Technically of course they don't, but for better understanding they should. Especially considering the fact that utils ResourceLoader is accessible from other modules.

> Anyway, I've renamed it to just Loader.

It's not really helpful. Now it looks like the ResourceLoader should implement Loader and Loader should be in utils module as well since it's general enough. I think in this case specifying what utils `ResourceLoader` does is a better way. `TileStoreResourceLoader`, for example.

Reply:
No

Reply:
What exactly are you negating here? And why?
What don't you like about my suggestion and what are the advantages of yours?